pkg/daemon/certificate_writer.go

```diff
@@ -186,7 +214,71 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 		}
 		klog.Infof("Certificate was synced from controllerconfig resourceVersion %s", controllerConfig.ObjectMeta.ResourceVersion)
 
+		if cmErr == nil {
+			cmd := exec.Command("systemctl", "stop", "kubelet")
+			errOut := &bytes.Buffer{}
+			cmd.Stderr = errOut
+			err = cmd.Run()
+			if err != nil {
+				return fmt.Errorf("Could not stop kubelet %s, %v", errOut.String(), err)
+			}
+
+			err = os.Remove("/var/lib/kubelet/kubeconfig")
+			if err != nil {
+				return fmt.Errorf("Could not remove kubelet's kubeconfig file: %v", err)
+			}
+
+			cmd = exec.Command("systemctl", "start", "kubelet")
+			errOut = &bytes.Buffer{}
+			cmd.Stderr = errOut
+			err = cmd.Run()
+			if err != nil {
+				return fmt.Errorf("Could not start kubelet %s, %v", errOut.String(), err)
+			}
+		}
+
 	}
 
 	return nil
 }
+
+func cmToData(cm *corev1.ConfigMap, key string) ([]byte, error) {
+	if bd, bdok := cm.BinaryData[key]; bdok {
+		return bd, nil
+	} else if d, dok := cm.Data[key]; dok {
```

Comment:
suggestion: Drop the `else` here and put `if d, ok //...` on a new line. The logic is a bit easier to parse that way.